Remotion/Validation/UnitTests/RoleCollectors/AddingPropertyValidationRuleCollector1Test.cs

```diff
@@ -105,51 +124,155 @@ public void Create_PropertyDeclaredInBaseClass ()
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void Create_BuildsFuncForPropertyAccess ()
     {
+      var type = new Customer() { UserName = "Test" };
+
+      Assert.That(_addingPropertyValidationRuleCollectorType.PropertyFunc(type), Is.EqualTo("Test"));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationRule_InitializesDeferredInitializationValidationMessages ()
     {
+      var validationMessage = new DelegateBasedValidationMessage(() => "Test");
+
+      var validationMessageMock = new Mock<IValidationMessageFactory>();
+      validationMessageMock
+          .Setup(e => e.CreateValidationMessageForPropertyValidator(It.IsAny<NotNullValidator>(), _property))
+          .Returns(validationMessage);
+
+      Func<PropertyValidationRuleInitializationParameters, IPropertyValidator> validatorFactory = param => new NotNullValidator(param.ValidationMessage);
+
+      _addingPropertyValidationRuleCollector.RegisterValidator(validatorFactory);
+      var result = _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock.Object);
+
+      var notNullValidators = ((IPropertyValidationRule)result).Validators.Cast<NotNullValidator>().ToArray();
+      Assert.That(notNullValidators.Length, Is.EqualTo(1));
+      Assert.That(notNullValidators[0].ValidationMessage.ToString(), Is.EqualTo("Test"));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationRule_WhenCalledTwice_InitializesDeferredInitializationValidationMessagesOnlyForNewlyRegisteredValidators ()
     {
+      var validationMessage1 = new DelegateBasedValidationMessage(() => "Test");
+      var validationMessage2 = new DelegateBasedValidationMessage(() => "newTest");
+
+      var validationMessageMock1 = new Mock<IValidationMessageFactory>();
```

Comment:
the name of the mock is incorrect. it should be a) called a `stub` and be this is `validationMessageFactory` not a `validationMessage`. Please check the other tests, too.